connect/zypper_test.go

```diff
@@ -1,29 +1,19 @@
 package connect
 
 import (
+	_ "embed"
 	"fmt"
 	"os"
 	"os/exec"
 	"testing"
 )
 
-var testProducts = `<?xml version='1.0'?>
-<stream>
-  <product-list>
-    <product name="SUSE-MicroOS" version="5.0" release="1" epoch="0" arch="x86_64" vendor="SUSE" summary="SUSE Linux Enterprise Micro 5.0" repo="@System" productline="SUSE-MicroOS" registerrelease="" shortname="SUSE Linux Enterprise Micro" flavor="" isbase="true" installed="true"><endoflife time_t="1606694400" text="2020-11-30T00:00:00Z"/><registerflavor/><description>SUSE Linux Enterprise Micro 5.0</description></product>
-    <product name="suse-openstack-cloud" version="8" release="0" epoch="0" arch="x86_64" vendor="SUSE LINUX GmbH, Nuernberg, Germany" summary="SUSE OpenStack Cloud 8" repo="@System" productline="suse-openstack-cloud" registerrelease="" shortname="SOC8" flavor="POOL" isbase="false" installed="true"><endoflife time_t="1622419200" text="2021-05-31T00:00:00+00"/><registerflavor>extension</registerflavor><description>SUSE OpenStack Cloud 8</description></product>
-  </product-list>
-</stream>
-`
-
-var testProductsNoBase = `<?xml version='1.0'?>
-<stream>
-  <product-list>
-    <product isbase="false" name="SUSE-MicroOS" version="5.0" release="1" epoch="0" arch="x86_64" vendor="SUSE" summary="SUSE Linux Enterprise Micro 5.0" repo="@System" productline="SUSE-MicroOS" registerrelease="" shortname="SUSE Linux Enterprise Micro" flavor="" installed="true"><endoflife time_t="1606694400" text="2020-11-30T00:00:00Z"/><registerflavor/><description>SUSE Linux Enterprise Micro 5.0</description></product>
-    <product isbase="false" name="suse-openstack-cloud" version="8" release="0" epoch="0" arch="x86_64" vendor="SUSE LINUX GmbH, Nuernberg, Germany" summary="SUSE OpenStack Cloud 8" repo="@System" productline="suse-openstack-cloud" registerrelease="" shortname="SOC8" flavor="POOL" installed="true"><endoflife time_t="1622419200" text="2021-05-31T00:00:00+00"/><registerflavor>extension</registerflavor><description>SUSE OpenStack Cloud 8</description></product>
-  </product-list>
-</stream>
-`
+var (
+	//go:embed testdata/products.xml
+	testProducts string
+	//go:embed testdata/products-no-base.xml
+	testProductsNoBase string
```

Comment:
I'm not sure about moving testdata into the packager directory. I think it's more conventional to have it outside. Maybe we don't need  to use go:embed for test binaries if we do something like this https://github.com/SUSE/connect-ng/pull/10

Reply:
#10 looks great! It solves the problem of test data in the code (which doesn't have syntax coloring, is hard to compare samples and which is generally ugly) and saves us trouble of inventing new names for test variables every time we get a new test file. I'll rebase once we have that merged.

Reply:
done